src/support.cpp

```diff
@@ -783,6 +788,213 @@ void AreaSupport::generateOverhangAreasForMesh(SliceDataStorage& storage, SliceM
                                });
 }
 
+Polygons AreaSupport::generateVaryingXYDisallowedArea(const SliceMeshStorage& storage, const Settings& infill_settings, const LayerIndex layer_idx)
+{
+    const auto& mesh_group_settings = Application::getInstance().current_slice->scene.current_mesh_group->settings;
+    const auto layer_thickness = mesh_group_settings.get<coord_t>("layer_height");
+    const auto support_distance_top = mesh_group_settings.get<coord_t>("support_top_distance");
+    const auto support_distance_bot = mesh_group_settings.get<coord_t>("support_bottom_distance");
```

Comment:
```suggestion
    const auto support_distance_top = static_cast<double>(mesh_group_settings.get<coord_t>("support_top_distance"));
        const auto support_distance_bot = static_cast<double>(mesh_group_settings.get<coord_t>("support_bottom_distance"));
```

you're only using it as a double everywhere